tests/test_api.rs

```diff
@@ -2930,20 +2946,34 @@ fn snapshot_creator() {
 }
 
 lazy_static! {
-  static ref EXTERNAL_REFERENCES: v8::ExternalReferences =
-    v8::ExternalReferences::new(&[v8::ExternalReference {
-      function: fn_callback.map_fn_to()
-    }]);
+  static ref EXTERNAL_REFERENCES: Mutex<v8::ExternalReferences> =
+    Mutex::new(v8::ExternalReferences::new(&[
+      v8::ExternalReference {
+        function: fn_callback.map_fn_to()
+      },
+      v8::ExternalReference {
+        function: fn_callback_external.map_fn_to()
+      }
+    ]));
 }
 
 #[test]
 fn external_references() {
   let _setup_guard = setup();
+  let mut refs_mut = EXTERNAL_REFERENCES.lock().unwrap();
+  // Allocate externals for the test.
+  let external_ptr = Box::into_raw(vec![0_u8, 1, 2, 3, 4].into_boxed_slice())
+    as *mut [u8] as *mut c_void;
+  // Push them to the external reference table.
+  refs_mut.push(v8::ExternalReference {
+    pointer: external_ptr,
+  });
+  let refs: &'static v8::ExternalReferences =
+    Box::leak(Box::new(refs_mut.clone()));
```

Comment:
Why does this need to be leaked?

Reply:
Because of the `'static` lifetime requirement on the [`Allocated`](https://github.com/denoland/rusty_v8/blob/main/src/support.rs#L375) type.